eng/pipelines/templates/steps/set-artifact-packages.yml

```diff
@@ -1,7 +1,19 @@
 parameters:
   PackageInfo: ''
   Artifacts: []
+  ServiceDirectory: ''
 steps:
+  - ${{ if ne(parameters.ServiceDirectory, 'auto') }}:
+    - pwsh: |
+        $artifacts = '${{ convertToJson(parameters.Artifacts) }}' | ConvertFrom-Json
+
+        if (!$artifacts) {
```

Comment:
I would suggest just handling this in the Resolve Targeted Packages steps if the pkgNames is empty. That way you can dump any artifacts found and any listed in the parameters for better debugging. Plus, it will handle cases where the filters don't find any matches as well.

Reply:
> and any listed in the parameters for better debugging

It'll be non in the parameters for the release build, which is the issue. I'll still do what you're asking though. Just adjusted for necessity.